control_panel_api/serializers.py

```diff
@@ -161,3 +145,43 @@ def update(self, instance, validated_data):
             )
 
         return super().update(instance, validated_data)
+
+
+class UserUserAppSerializer(serializers.ModelSerializer):
+    """Used from within with UserSerializer to explicitly expose the app
+    but hide the User
+    """
+    app = SimpleAppSerializer()
+
+    class Meta:
+        model = UserApp
+        fields = ('id', 'app', 'is_admin')
+
+
+class UserUserS3BucketSerializer(serializers.ModelSerializer):
+    """Used from within with UserSerializer to explicitly expose the s3bucket
+    but hide the User
+    """
+    s3bucket = SimpleS3BucketSerializer()
+
+    class Meta:
+        model = UserS3Bucket
+        fields = ('id', 's3bucket', 'access_level', 'is_admin')
+
+
+class UserSerializer(serializers.ModelSerializer):
+    userapps = UserUserAppSerializer(many=True, read_only=True)
```

Comment:
Using possible new names this would become:

```python
userapps = UserAppNestedInUserSerializer(many=True, read_only=True)
users3buckets = UserS3BucketNestedInUserSerializer(many=True, read_only=True)
```